### PR TITLE
Add additional methods to BitFlags<State>

### PR DESCRIPTION
**Pull request by TTWNO**
I should probably add all those other methods too from [enumflags2](https://docs.rs/enumflags2/latest/enumflags2/struct.BitFlags.html) so that it is truly transparent.

Please await further commits

### CODE REVIEW

**Comment by TTWNO** [comment]
> Ideally such a commit would add a test too.

I'm just passing through info to BitFlags, not really doing any original work. Unless `enumflags2` is having issues, we shouldn't be either.

File: src/state.rs [hunk 1]
```diff
@@ -271,6 +276,31 @@ impl StateSet {
     pub fn iter(self) -> impl Iterator<Item = State> {
         self.0.iter()
     }
+
+		/// Create a StateSet, with all states active.
+		pub fn all() -> Self {
+			StateSet(State::all())
+		}
+
+		/// Checks if all states are set.
```
Comment (DataTriny) [inline src/state.rs:285]: I don't see any use case for this. The purpose of creating newtypes is to guide people into using the API correctly, as well as giving them the tools that make sense. A `StateSet` is not a `BitFlags`. Therefore not all methods of `BitFlags` are applicable to `StateSet`.

File: src/state.rs [hunk 2]
```diff
@@ -271,6 +276,31 @@ impl StateSet {
     pub fn iter(self) -> impl Iterator<Item = State> {
         self.0.iter()
     }
+
+		/// Create a StateSet, with all states active.
```
Comment (DataTriny) [inline src/state.rs:280]: I am not sure why would anyone need this.

**Review by DataTriny — CHANGES_REQUESTED** [review]

File: src/state.rs (hunk 2 above)
Comment (TTWNO) [reply src/state.rs:280]: Even just as a debug mechanism, showing a number or a bitset would be a lot less clear than a list of the states.

**Comment by TTWNO** [comment]
I've removed some of the bitset methods that don't make as much sense, like is_all, and all(), but I've left the rest since the usage of those seems fairly clear.

File: src/state.rs [hunk 3]
```diff
@@ -344,6 +374,41 @@ impl std::ops::BitXor for StateSet {
         StateSet(self.0 ^ other.0)
     }
 }
+impl std::ops::BitXorAssign for StateSet {
```
[the remark this answers is not in the document]
Comment (DataTriny) [reply src/state.rs:367]: Can you please add a blank line between each `impl` block?

File: src/state.rs [hunk 4]
```diff
@@ -262,6 +262,11 @@ impl StateSet {
         self.0.contains(other)
     }
 
+		/// Removes a [`State`] (optionally) previously contained in the [`StateSet`].
```
Comment (DataTriny) [inline src/state.rs:265]: I don't like having links pointing to the current item inside its documentation. You already know you are reading the documentation for `StateSet`. Therefore it's only noise, especially for screen reader users.

**Review by DataTriny — CHANGES_REQUESTED** [review]

**Review by DataTriny — APPROVED** [review]

**Pull request merged**